tar/chunks/chunks_directory.c

```diff
@@ -362,6 +362,48 @@ chunks_directory_write(const char * cachepath, RWHASHTAB * HT,
 }
 
 /**
+ * chunks_directory_exists(cachepath):
+ * Return 1 if the /directory file exists within ${cachepath}, 0 if it does
+ * not, or -1 if there is an error.
+ */
+int
+chunks_directory_exists(const char * cachepath)
+{
+	char * directory_filename;
+	struct stat sb;
+
+	/* Prepare filename. */
+	if (asprintf(&directory_filename, "%s/directory", cachepath) == -1)
+		goto err0;
+
+	/* Check if file exists. */
+	if (stat(directory_filename, &sb) == 0)
+		goto fileexists;
+	else {
+		if (errno != ENOENT)
+			goto err1;
+	}
+
```

Comment:
Given the three-way replication of freeing `directory_filename`, I'm tempted to break the usual style rule of "one return() statement per meaning" here.  `int rc;` which gets set to 0/1/-1, then `/* Return result code. */` `return (rc);`...